nodes/src/nodes/multi_agent/agent_definition.py

```
@@ -0,0 +1,120 @@
+    def from_dict(cls, data: Dict[str, Any]) -> 'AgentDefinition':
+        """Create an AgentDefinition from a raw JSON-parsed dictionary.
+
+        Validates that no unexpected keys are present to prevent
+        arbitrary data injection.
+
+        Raises:
+            ValueError: If *data* is not a dict or contains unknown keys.
+            TypeError: If field types are wrong.
+        """
+        if not isinstance(data, dict):
+            raise ValueError(f'Agent definition must be a dict, got {type(data).__name__}')
```

Ruff (0.15.7)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)